openfe/utils/visualization.py

```diff
@@ -140,22 +157,49 @@ def draw_mapping(mol1_to_mol2: Dict[int, int],
     if not d2d:
         d2d = Chem.Draw.rdMolDraw2D.MolDraw2DCairo(600, 300, 300, 300)
 
-    # Need to squash to 2D
-    mol1_copy = copy.deepcopy(mol1)
-    mol2_copy = copy.deepcopy(mol2)
-    AllChem.Compute2DCoords(mol1_copy)
-    AllChem.Compute2DCoords(mol2_copy)
+    return _draw_molecules(
+        d2d,
+        [mol1, mol2],
+        atoms_list=atoms_list,
+        bonds_list=bonds_list,
+        atom_colors=atom_colors,
+        highlight_color=red,
+    )
 
-    # Use the d2d object we instantiated or the one passed in by the user
-    d2d.drawOptions().useBWAtomPalette()
-    d2d.drawOptions().continousHighlight = False
-    d2d.drawOptions().setHighlightColour(red)
-    d2d.drawOptions().addAtomIndices = True
-    d2d.DrawMolecules(
-        [mol1_copy, mol2_copy],
-        highlightAtoms=atoms_list,
-        highlightBonds=bonds_list,
-        highlightAtomColors=atom_colors,
+
+def draw_one_molecule_mapping(mol1_to_mol2, mol1, mol2, d2d=None):
+    """Draw the mapping visualization for a single molecular of a mapping
+
+    This will always draw ``mol1``. To draw ``mol2``, switch order/invert
+    ``mol1_to_mol2`` mapping.
+    """
+    uniques = _get_unique_bonds_and_atoms(mol1_to_mol2, mol1, mol2)
+    atoms_list = [uniques["atoms"] | uniques["elements"]]
+    bonds_list = [uniques["bonds"]]
+    red = (220/255, 50/255, 32/255, 1)
+    blue = (0, 90/255, 181/255, 1)
+
+    atom_colors = [{at: blue for at in uniques["elements"]}]
+
+    if d2d is None:
+        # TODO: we should try to be smarter about the default size here
```

Comment:
I think it's probably best not to overthink this one, if folks want a specific resolution / size they should just provide their own d2d object?

Reply:
Agreed. I think some of this came from my matplotlib-specific scratch work. I was thinking about the fact that a lot of stuff in matplotlib is given in inches, whereas rdkit is taking things in pixels -- it could be nice to use the existing matplotlib figure to do something smart here.

On reflection, I think that goes beyond the scope of these methods -- they don't need to know the existence of matplotlib. Comment removed (and relevant code moved to `_draw_molecules` as discussed in another comment).